restalchemy/api/controllers.py

```diff
@@ -427,6 +427,60 @@ def update(self, parent_resource, uuid, **kwargs):
         return dm
 
 
+class PaginationCursor:
+    """
+    Represents a pagination cursor with sort value and ID.
+    Handles non-trivial logic for paginating by non-unique column.
+    """
+
+    def __init__(
+        self, model, marker_id, sort_column=None, sort_direction="asc"
+    ):
+        """
+        Create pagination cursor from marker ID.
+
+        Args:
+            model: The model to query
+            marker_id: The ID of the last row from previous page
+            sort_column: Column name to sort by (None for ID-only sort)
+            sort_direction: "asc" or "desc"
+        """
+        self.marker_id = marker_id
+        self.sort_col = sort_column
+        self.sort_dir = sort_direction
+        self.id_name = model.get_id_property_name()
+        self._fetch_sort_val(model)
```

Comment:
Nit - I propose to make it explicit rather than implicit and return value from _fetch_sort_val and set it as we want, like
```suggestion
        self.sort_value = self._fetch_sort_val(model)
```

(and appropriate change in _fetch_sort_val)